hummingbot/connector/exchange/xrpl/xrpl_order_book.py

```diff
@@ -0,0 +1,107 @@
+from typing import Dict, Optional
+
+from xrpl.utils import drops_to_xrp
+
+from hummingbot.connector.exchange.xrpl.xrpl_utils import normalize_price_from_drop
+from hummingbot.core.data_type.order_book import OrderBook
+from hummingbot.core.data_type.order_book_message import OrderBookMessage, OrderBookMessageType
+from hummingbot.core.data_type.order_book_row import OrderBookRow
+
+
+class XRPLOrderBook(OrderBook):
+
+    @classmethod
+    def snapshot_message_from_exchange(cls,
+                                       msg: Dict[str, any],
+                                       timestamp: float,
+                                       metadata: Optional[Dict] = None) -> OrderBookMessage:
+        """
+        Creates a snapshot message with the order book snapshot message
+        :param msg: the response from the exchange when requesting the order book snapshot
+        :param timestamp: the snapshot timestamp
+        :param metadata: a dictionary with extra information to add to the snapshot data
+        :return: a snapshot message with the snapshot information received from the exchange
+        """
+        is_xrp_pair = False
+
+        if metadata:
+            msg.update(metadata)
+            trading_pair = metadata["trading_pair"]
+            # split trading pair to get token, then check if token is XRP
+            is_xrp_pair = trading_pair.split("-")[1] == "XRP" or trading_pair.split("-")[0] == "XRP"
```

Comment:
Consider not hardcoding the native asset to that the connector can also be used with the Xahau Network

Reply:
Hey @Jbekker, this is to handle special case where "XRP" is denominated in drops on XRP Ledger which require additional calculation. Is that different on Xahau?

Reply:
It will be the same for Xahau but then the native asset/currency is 'XAH'

Reply:
Thanks, I will keep this in mind for next updates.